File: backend/src/main/java/com/back/team11/domain/cafe/dto/PageResponse.java
package com.back.team11.domain.cafe.dto;

import org.springframework.data.domain.Page;
import java.util.List;

///  관리자 카페 목록 조회 시 페이징된 정보들을 보여줄 때 꼭 필요한 필수정보들만 표현
public record PageResponse<T>(
        List<T> content,       // 실제 데이터 목록
        int currentPage,       // 현재 페이지 번호 (1부터 시작)
        int totalPages,        // 전체 페이지 수
        long totalElements     // 전체 데이터 개수
) {
    // Spring의 Page 객체를 PageResponse로 변환하는 정적 팩토리 메서드
    public static <T> PageResponse<T> of(Page<T> page) {
        return new PageResponse<>(
                page.getContent(),
                page.getNumber() + 1, // 클라이언트 친화적으로 0-based index를 1-based index로 변경
                page.getTotalPages(),
                page.getTotalElements()
        );
    }
}

File: backend/src/main/java/com/back/team11/domain/cafe/dto/PageResponse.kt
package com.back.team11.domain.cafe.dto

import org.springframework.data.domain.Page

/**  관리자 카페 목록 조회 시 페이징된 정보들을 보여줄 때 꼭 필요한 필수정보들만 표현 */
data class PageResponse<T>(
    val content: List<T>,  // 실제 데이터 목록
    val currentPage: Int,  // 현재 페이지 번호 (1부터 시작)
    val totalPages: Int,  // 전체 페이지 수
    val totalElements: Long // 전체 데이터 개수
) {
    companion object {
        // Spring의 Page 객체를 PageResponse로 변환하는 정적 팩토리 메서드
        fun <T : Any> of(page: Page<T>): PageResponse<T> = PageResponse(
            content = page.content,
            currentPage = page.number + 1, // 0-based index를 1-based index로 변경
            totalPages = page.totalPages,
            totalElements = page.totalElements,
        )
    }
}